Source/WebCore/rendering/RenderReplaced.cpp

```diff
@@ -489,7 +497,7 @@ void RenderReplaced::computeIntrinsicRatioInformation(FloatSize& intrinsicSize,
 
     if (style().hasAspectRatio()) {
         intrinsicRatio = style().logicalAspectRatio();
-        if (style().aspectRatioType() == AspectRatioType::Ratio)
+        if (auto* video = dynamicDowncast<RenderVideo>(this); style().aspectRatioType() == AspectRatioType::Ratio || (video && video->hasDefaultObjectSize()))
```

Comment:
Then you could use that helper function here:

```
if (style().aspectRatioType() == AspectRatioType::Ratio || isVideoWithDefaultObjectSize(this)) {
    ....
}
```

Reply:
Changed